File: exJava/go/Q15.go
package main

import "fmt"

func main() {
  newFeature();
}

type Feature struct {
  animal Animal
}
func newFeature() (*Feature) {
  var line string
  fmt.Printf("input: ")
  fmt.Scanf("%s\n", &line)

  f := new(Feature)
  f.animal = new(Creator).create(line) 
  f.check()
  return f
}
func (f *Feature) check() {
  if f.animal == nil {
    fmt.Printf("Error")
  } else {
    fmt.Printf("Kind: %s\n", (f.animal).getKind())
    fmt.Printf("Move: %s\n", (f.animal).getMove())
    fmt.Printf("Child: %s\n", (f.animal).getHowChild())
    fmt.Printf("Die: %t\n", (f.animal).willDie())
  }
}

type Creator struct { }
func (_ Creator) create (line string) Animal {
  if line == "inu" {
    return new(Dog)
  } else if line == "bird" {
    return new(Bird)
  } else if line == "human" {
    return new(Human)
  } else {
    return nil 
  }
}

type Animal interface {
  getKind() string
  getMove() string
  willDie() bool
  getHowChild() string
}

type Mammal struct { } 
func (_ Mammal) getKind() string {
  return "Animal"
}
func (_ Mammal) getMove() string {
  return "Move on Foot"
}
func (_ Mammal) willDie() bool {
  return true 
}
func (_ Mammal) getHowChild() string {
  return "as Child"
}

type Human struct {
  Mammal
} 
func (_ Human) getKind() string {
  return "Human"
}
func (_ Human) getMove() string {
  return "Two legs walking"
}

type Dog struct {
  Mammal
} 
func (_ Dog) getKind() string {
  return "Dog"
}
func (_ Dog) getMove() string {
  return "Four legs walking"
}

type Bird struct { } 
func (_ Bird) getKind() string {
  return "Bird"
}
func (_ Bird) getMove() string {
  return "Fly"
}
func (_ Bird) willDie() bool {
  return true 
}
func (_ Bird) getHowChild() string {
  return "as Egg"
}

File: shibaura-institute-technology/exJava/java/Q15.java
import java.io.*;

public class Q5 {
  public static void main (String args[]) {
    new Feature();
  }
}

class Feature {
  Animal animal;

  public Feature() {
    String line;
    BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
    try {
      System.out.print("\ninput: "); 
      line = reader.readLine();
      animal = new Creator().create(line);
    } catch (IOException e) {
      System.err.println(e.getMessage());
    }
    check();
  }

  public void check() {
    if (animal == null) {
      System.out.println("Error"); 
    } else {
      System.out.print("Kind: "); 
      System.out.println(animal.getKind()); 
      System.out.print("Move: "); 
      System.out.println(animal.getMove()); 
      System.out.print("Child: "); 
      System.out.println(animal.getHowChild()); 
      System.out.print("Die: "); 
      System.out.println(animal.willDie()); 
    }
  }
}

class Creator {
  public Animal create(String line) {
    if (line.equals("inu")) {
      return new Dog();
    } else if (line.equals("bird")) {
      return new Bird();
    } else if (line.equals("human")) {
      return new Human();
    } else {
      return null; 
    }
  } 
}

abstract class Animal {
 public abstract String getKind();
 public abstract String getMove();
 public abstract Boolean willDie();
 public abstract String getHowChild();
}

class Mammal extends Animal {
  public String getKind() {
    return "Animal";
  }
  public String getMove() {
    return "Move on Foot";
  }
  public Boolean willDie() {
    return true;
  }
  public String getHowChild() {
    return "as Child";
  }
}

class Human extends Mammal {
  public String getKind() {
    return "Human";
  }
  public String getMove() {
    return "Two legs walking";
  }
}

class Dog extends Mammal {
  public String getKind() {
    return "Dog";
  }
  public String getMove() {
    return "Four legs walking";
  }
}

class Bird extends Animal {
  public String getKind() {
    return "Bird";
  }
  public String getMove() {
    return "Fly";
  }
  public Boolean willDie() {
    return true;
  }
  public String getHowChild() {
    return "as Egg";
  }
}
